### DISCUSSION TITLE
No module named 'checksumdir'

### DISCUSSION THREAD

**Discussion by Nekuromu** [pyinstaller/pyinstaller#7424] [Help]
I just started coding in python about 2 months back, I am making a save backup utility for Kenshi (due to a save deleting bug they have ongoing).

The Error:
```
Traceback (most recent call last):
  File "main.py", line 1, in <module>
  File "PyInstaller\loader\pyimod02_importers.py", line 499, in exec_module
  File "file_manager.py", line 3, in <module>
ModuleNotFoundError: No module named 'checksumdir'
```

The Pycharm project is comprised of 2 scripts, main.py and file_manager.py each one has a few imports.

```
main.py imports: 
from file_manager import FileManager
import os
import json
```

file_manager.py imports: 
```
import json
import shutil
from checksumdir import dirhash
import os
import datetime as dt
import py7zr
```

When programming I made sure to download the packages as they become underlined in red due to not being installed yet (freshly written out import need). "checksumdir" is no different. I use it in a single place in the FileManager class in file_manager.py. 
```
    def hash_dir(self, directory):
        return dirhash(directory)
```

I've tried tinkering with the pyinstaller parameters such as "onefile", or "onedir". Similarly I have tried using hidden-import and paths. None of them seem to resolve the missing checksumdir module.

`pyinstaller --windowed --clean --noconfirm main.py`
 is the current exe dist command line.

**Comment by Nekuromu** [comment]
I have my warn-main.txt from the build 
```
Types if import:
* top-level: imported at the top-level - look at these first
* conditional: imported within an if-statement
* delayed: imported within a function
* optional: imported within a try-except-statement

IMPORTANT: Do NOT post this list to the issue-tracker. Use it as a basis for
            tracking down the missing module yourself. Thanks!

missing module named 'org.python' - imported by copy (optional)
missing module named org - imported by pickle (optional)
missing module named pwd - imported by posixpath (delayed, conditional, optional), shutil (delayed, optional), tarfile (optional), pathlib (delayed, optional), subprocess (delayed, conditional, optional), netrc (delayed, conditional), getpass (delayed)
missing module named grp - imported by shutil (delayed, optional), tarfile (optional), pathlib (delayed, optional), subprocess (delayed, conditional, optional)
missing module named posix - imported by posixpath (optional), shutil (conditional), importlib._bootstrap_external (conditional), os (conditional, optional)
missing module named resource - imported by posix (top-level)
missing module named _frozen_importlib_external - imported by importlib._bootstrap (delayed), importlib (optional), importlib.abc (optional)
excluded module named _frozen_importlib - imported by importlib (optional), importlib.abc (optional)
missing module named _scproxy - imported by urllib.request (conditional)
missing module named termios - imported by getpass (optional)
missing module named _posixsubprocess - imported by subprocess (conditional)
missing module named fcntl - imported by subprocess (optional)
missing module named py7zr - imported by file_manager (top-level)
missing module named checksumdir - imported by file_manager (top-level)
```
this actually shows I am missing both checksumdir and py7zr
which both are inside my venv and inside the project packages
I've not had any issues with pyinstaller not picking up on installed packages so
I have no idea where to go from here

**Reply by bwoodsend** [reply]
[Sigh] Another PyCharm user falling victim to PyCharm's not so helpful environment meddling.

In a terminal with your virtual environment activated, what happens if you run:
```
python main.py
```
and
```
python -m PyInstaller main.py
```
I'm guessing that PyInstaller is installed in your main environment instead of your venv one so the second command will say *no module named PyInstaller*.
